cmd/root.go

```diff
@@ -58,27 +73,34 @@ func Execute() {
 }
 
 func init() {
-	// cobra.OnInitialize(initConfig)
-
-	// RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.namerctl.yaml)")
-
-	RootCmd.PersistentFlags().StringVar(&baseURLString, "base-url",
-		os.Getenv("NAMERCTL_BASE_URL"),
+	//jww.SetStdoutThreshold(jww.LevelTrace)
+	cobra.OnInitialize(initConfig)
+	RootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file")
+	RootCmd.PersistentFlags().StringVar(&baseURLString, "base-url", "",
 		"namer location (e.g. http://namerd.example.com:4080)")
+	viper.BindPFlag("base-url", RootCmd.PersistentFlags().Lookup("base-url"))
+}
+
+func addParentConfigPaths(dir string) {
+	viper.AddConfigPath(dir + string(os.PathSeparator))
+	if sep := strings.LastIndex(dir, string(os.PathSeparator)); sep != -1 {
+		addParentConfigPaths(dir[:sep])
+	}
 }
 
-//TODO
 // initConfig reads in config file and ENV variables if set.
-// func initConfig() {
-// 	if cfgFile != "" { // enable ability to specify config file via flag
-// 		viper.SetConfigFile(cfgFile)
-// 	}
-// 	viper.SetConfigName(".namerctl") // name of config file (without extension)
-// 	viper.AddConfigPath("$PWD")     // adding current directory as first search path
-// 	viper.AddConfigPath("$HOME")    // adding home directory as second search path
-// 	viper.AutomaticEnv()            // read in environment variables that match
-// 	// If a config file is found, read it in.
-// 	if err := viper.ReadInConfig(); err == nil {
-// 		fmt.Println("Using config file:", viper.ConfigFileUsed())
-// 	}
-// }
+func initConfig() {
+	if cfgFile != "" { // enable ability to specify config file via flag
+		viper.SetConfigFile(cfgFile)
+	}
+	viper.SetConfigName(".namerctl") // name of config file (without extension)
+	addParentConfigPaths(os.Getenv("PWD"))
+	viper.SetEnvPrefix("namerctl")
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
```

Comment:
Yeah, good question.  For some reason this is the code that cobra generates for you...  I don't like when commandline tools print needless (irregular) messages, so i'll remove this and add errors.